sdk/src/wallet/operations/transaction/high_level/staking/begin.rs

```diff
@@ -0,0 +1,101 @@
+    wallet::{types::TransactionWithMetadata, TransactionOptions, Wallet},
+};
+
+/// Parameters for beginning a staking period.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BeginStakingParams {
+    /// The account id which will become a validator.
+    pub account_id: AccountId,
+    /// The amount of tokens to stake.
+    pub staked_amount: u64,
+    /// The fixed cost of the validator, which it receives as part of its Mana rewards.
+    pub fixed_cost: u64,
+    /// The staking period (in epochs). Will default to the staking unbonding period.
+    pub staking_period: Option<u32>,
+}
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    pub async fn begin_staking(
+        &self,
+        params: BeginStakingParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let options = options.into();
+        let prepared = self.prepare_begin_staking(params, options.clone()).await?;
+
+        self.sign_and_submit_transaction(prepared, None, options).await
+    }
+
+    /// Prepares the transaction for [Wallet::begin_staking()].
+    pub async fn prepare_begin_staking(
+        &self,
+        params: BeginStakingParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<PreparedTransactionData> {
+        log::debug!("[TRANSACTION] prepare_begin_staking");
+
+        let account_id = params.account_id;
+        let account_output_data = self
+            .data()
+            .await
+            .unspent_account_output(&account_id)
+            .cloned()
+            .ok_or_else(|| crate::wallet::Error::AccountNotFound)?;
+
+        if account_output_data
+            .output
+            .features()
+            .map_or(false, |f| f.staking().is_some())
+        {
+            return Err(crate::wallet::Error::StakingFailed(format!(
+                "account id {account_id} already has a staking feature"
+            )));
+        }
+
+        let protocol_parameters = self.client().get_protocol_parameters().await?;
+
+        if let Some(staking_period) = params.staking_period {
+            if staking_period < protocol_parameters.staking_unbonding_period() {
+                return Err(crate::wallet::Error::StakingFailed(format!(
+                    "staking period {staking_period} is less than the minimum {}",
+                    protocol_parameters.staking_unbonding_period()
+                )));
+            }
+        }
+
+        let slot_commitment_id = self.client().get_issuance().await?.latest_commitment.id();
+        let past_bounded_epoch =
+            protocol_parameters.epoch_index_of(protocol_parameters.past_bounded_slot(slot_commitment_id));
+        let end_epoch = past_bounded_epoch
+            + params
+                .staking_period
```

Comment:
Don't we want u64::MAX here? When we want to stop, we can then reduce it

Reply:
Not sure! I mean, I think it's valid to do that but I don't know if we should or not

Reply:
I think I read it in some TIP that it's recommended, because you probably don't know when you want to stop staking and then you just need to update to stop

Reply:
Found the section https://github.com/iotaledger/tips/blob/tip42/tips/TIP-0042/tip-0042.md#expiry-slot-and-end-epoch